### PR TITLE
[Hub Generated] Publish private branch 'RPSaaSMaster-easm-data-plane'

### PR DESCRIPTION
**Pull request by soniaab**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/published%2FAzure%2Fazure-rest-api-specs-pr%2FRPSaaSMaster-easm-data-plane...main/).
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Comment by soniaab** [comment]
The PR is created based on the updates in the private branch. The updates in the PR has already been reviewed and approved with this PR [Azure/azure-rest-api-specs-pr/7043](https://github.com/Azure/azure-rest-api-specs-pr/pull/7043)

**Comment by eqtip** [comment]
@markweitzel @mikekistler @jhendrixMSFT Would we be able to get some movement on this?

**Comment by eqtip** [comment]
@markweitzel @mikekistler @jhendrixMSFT Another ping ...

**Comment by mikekistler** [comment]
Added @heaths and @MushMal as they approved the PR in the internal repo.

File: custom-words.txt
Comment (heaths) [inline custom-words.txt:4]: Why are you deleting anything from this file? I would expect your changes to only be additive, if necessary.

File: specification/riskiq/data-plane/Microsoft.Easm/preview/2022-04-01-preview/easm.json
Comment (heaths) [inline specification/riskiq/data-plane/Microsoft.Easm/preview/2022-04-01-preview/easm.json:1504]: We recommend enum values are camelCase, as they were prior.

File: specification/riskiq/data-plane/Microsoft.Easm/preview/2022-04-01-preview/easm.json
Comment (heaths) [inline specification/riskiq/data-plane/Microsoft.Easm/preview/2022-04-01-preview/easm.json:2259]: Why were all the descriptions removed?

File: specification/riskiq/data-plane/Microsoft.Easm/preview/2022-04-01-preview/easm.json
Comment (heaths) [inline specification/riskiq/data-plane/Microsoft.Easm/preview/2022-04-01-preview/easm.json:2361]: Why were all the descriptions removed?

Comment (heaths) [reply specification/riskiq/data-plane/Microsoft.Easm/preview/2022-04-01-preview/easm.json:2361]: Seems quite a few descriptions were removed from various models below.

**Review by heaths — APPROVED** [review]
Comparing with the prior def, I have a few minor comments but don't see anything holding up a preview.

File: custom-words.txt
Comment (eqtip) [reply custom-words.txt:4]: Shouldn't be.  Weird.  I'll look into it.

File: specification/riskiq/data-plane/Microsoft.Easm/preview/2022-04-01-preview/easm.json
Comment (eqtip) [reply specification/riskiq/data-plane/Microsoft.Easm/preview/2022-04-01-preview/easm.json:1504]: Yup caught and fixed this with the 2022-09-01-preview version.  We will be opening up a PR for that merge after this one is merged down. ;-)

File: specification/riskiq/data-plane/Microsoft.Easm/preview/2022-04-01-preview/easm.json
Comment (eqtip) [reply specification/riskiq/data-plane/Microsoft.Easm/preview/2022-04-01-preview/easm.json:2361]: Same as above.  Fixed in the 2022-09-01-preview version.

File: custom-words.txt
Comment (eqtip) [reply custom-words.txt:4]: Looks to me that this is a diff "issue" with that branch we started from and what we are targeting.  The changes should only include our updates which were only additions.  Can I just go ahead and fix it up on this branch?

Comment (heaths) [reply custom-words.txt:4]: It's best you rebase on main and force push. Simple merges get very messy:

```bash
# assuming upstream is Azure/azure-rest-api-specs and origin is a eqtip/azure-rest-api-specs fork (don't generally push to upstream and work in a fork instead)
git fetch upstream main
# on your published/Azure/azure-rest-api-specs-pr/RPSaaSMaster-easm-data-plane branch:
git rebase upstream/main
# fix any conflicts and follow on-screen instructions
git push --force
```

Comment (eqtip) [reply custom-words.txt:4]: @heaths So the problem here is that there are a number of changes coming in from our branch that are from its source branch that I believe we don't want applied to the public repo.  Specifically there are a TON of adds and removes we didn't make.  

By rebasing I could fix the conflicts but still ended up with said changes.  I instead created a branch that manually applied our changes and it's way more sensible:

https://github.com/Azure/azure-rest-api-specs/compare/main...eqtip:azure-rest-api-specs:published/Azure/azure-rest-api-specs-pr/RPSaaSMaster-easm-data-plane

Thoughts?

Comment (heaths) [reply custom-words.txt:4]: Anything in main is published and you cannot make breaking changes. Whether you merge (messy) or rebase (clean), you still end up with the same result (rebasing is just easier to review). There should be nothing in main you don't want.

All rebasing does is like taking all your commits from a common ancestor and effectively "replays" them (not quite, but a close analogy) on main after updating your branch. See https://git-scm.com/book/en/v2/Git-Branching-Rebasing

Comment (eqtip) [reply custom-words.txt:4]: @heaths Thanks ... I'm familiar with rebasing just have dealt with conflicts in different ways for different projects in the past and wanted to get the lay of the land.  With that said, I've push my changes to my fork:

https://github.com/Azure/azure-rest-api-specs/compare/main...eqtip:azure-rest-api-specs:published/Azure/azure-rest-api-specs-pr/RPSaaSMaster-easm-data-plane

Thanks look good but where to from here?  Do I need to open up a separate PR?  As far as I can tell I'm not allowed to push changes to the branch in the Azure repo.

Comment (heaths) [reply custom-words.txt:4]: You could force-push to this PR. But is this swagger considered _done_? The `main` branch is for publishing specs that, mostly, should be immutable after merging to `main`. See https://aka.ms/AzBreakingChangesPolicy for more information.

If not, others should make topic branches from this branch and submit PRs to it until your done, or make a dev- branch as described here: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/630/Walkthrough-Creating-a-new-service-specification-in-a-team

Comment (eqtip) [reply custom-words.txt:4]: Yes.  It is for this particular version.  We have another follow up version that went through the review process and will subsequently go through this same merge process.

Comment (eqtip) [reply custom-words.txt:4]: @heaths It doesn't look like I have permission to push to the PR branch.

Comment (eqtip) [reply custom-words.txt:4]: @heaths Any idea why I wouldn't have access to be able to push?  I've tried just about everything but no luck.  It should be noted I am not the original branch author.  Seems crazy that would be a limitation but maybe so?

Comment (heaths) [reply custom-words.txt:4]: What error did you get? It's not a protected branch.

Comment (eqtip) [reply custom-words.txt]: @heaths Figured it out.  Was missing membership in a particular team.  Updates have been pushed so conflicts are resolved.

**Pull request merged**
